src/lib/Query/Common/CriterionVisitor/MapLocation/MapLocationDistanceRange.php

```diff
@@ -88,6 +91,71 @@ public function visit(Criterion $criterion, CriterionVisitor $subVisitor = null)
 
         return '(' . implode(' OR ', $queries) . ')';
     }
-}
 
-class_alias(MapLocationDistanceRange::class, 'EzSystems\EzPlatformSolrSearchEngine\Query\Common\CriterionVisitor\MapLocation\MapLocationDistanceRange');
+    private function visitForSolr9(Criterion $criterion): string
+    {
+        if (is_array($criterion->value)) {
+            $minDistance = $criterion->value[0];
+            $maxDistance = $criterion->value[1] ?? 63510;
+        } else {
+            $minDistance = 0;
+            $maxDistance = $criterion->value;
+        }
+
+        $sign = '';
+        if (($criterion->operator === Operator::GT) ||
+            ($criterion->operator === Operator::GTE)) {
+            $sign = '-';
+        }
+
+        $searchFields = $this->getSearchFields(
+            $criterion,
+            $criterion->target,
+            $this->fieldTypeIdentifier,
+            $this->fieldName
+        );
+
+        if (empty($searchFields)) {
+            throw new InvalidArgumentException(
+                '$criterion->target',
+                "No searchable Fields found for the provided Criterion target '{$criterion->target}'."
+            );
+        }
+
+        /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion\Value\MapLocationValue $location */
+        $location = $criterion->valueData;
+
+        $queries = [];
+        foreach ($searchFields as $name => $fieldType) {
+            if ($criterion->operator === Operator::BETWEEN) {
+                $query = sprintf(
+                    '{!geofilt sfield=%s pt=%F,%F d=%s} AND -{!geofilt sfield=%s pt=%F,%F d=%s}',
+                    $name,
+                    $location->latitude,
+                    $location->longitude,
+                    $maxDistance,
+                    $name,
+                    $location->latitude,
+                    $location->longitude,
+                    $minDistance
+                );
+            } else {
+                $query = sprintf('%s{!geofilt sfield=%s pt=%F,%F d=%s}', $sign, $name, $location->latitude, $location->longitude, $maxDistance);
+            }
+
+            $queries[] = "{$query} AND {$name}:[* TO *]";
+        }
+
+        return '(' . implode(' OR ', $queries) . ')';
+    }
+
+    private function isSolrInMaxVersion(string $maxVersion): bool
+    {
+        $version = getenv('SOLR_VERSION');
+        if (is_string($version) && !empty($version)) {
+            return version_compare($version, $maxVersion, '<');
+        }
+
+        return true;
```

Comment:
You cannot rely on SOLR_VERSION env variable here. This will not work in production environment.